### PR TITLE
SIG-scale slos document

### PR DESCRIPTION
**Pull request by rthallisey**
Signed-off-by: Ryan Hallisey <rhallisey@nvidia.com>

Create and SLOs document for KubeVirt to describe how the community is testing Scale and Perf and what the current testing thresholds are for perf & scale. 

```release-note
NONE
```

### CODE REVIEW

**Comment by rthallisey** [comment]
cc @marceloamaral @davidvossel

File: docs/scale-and-perf/SLOs.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+but about the time to perform a batch of operations.  A good example of a Burst workload is a sudden spike in
+demand for compute resources to handle a massive increase of users.
+
+The **Burst Test** measures batch VM/VMI creation latency so here's how KubeVirt will test it.
+The Burst Test will ramp up object count in a datacenter, measure, then ramp down object count in a datacenter.
+The measurement will include the Object Creation rate (objects/second), VMI Phase Transition Rate, and a count for
+Kubernetes REST API calls.
+
+The **Steady State Test** takes a different approach than the burst test in that instead of starting the system
+from an unspecified capacity, the Steady State Test focuses on how well the system maintains its starting capacity.
```
Comment (marceloamaral) [inline docs/scale-and-perf/SLOs.md:19]: The **Steady State Test** takes a different approach to the Burst Test, showing the behavior of the system during normal operations under constant load to reach steady state shortly after the test load starts and ramp down to its previous state as soon as as the load is removed. For example, a system that has multiple users generating requests to create, update and delete VMs, in the steady-state phase the system will have an average number of operations per second, for example 20 create, update and delete/s.

File: docs/scale-and-perf/SLOs.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+For example, a system that maintains a certain capacity of warm resources (an active resource that isn't being used by a customer)
```
Comment (marceloamaral) [inline docs/scale-and-perf/SLOs.md:20]: For example, users will create multiple objects, apply updates, and eventually delete objects, generating an average number of operations per second.

**Review by marceloamaral — CHANGES_REQUESTED** [review]
Look good! I have few comment.

File: docs/scale-and-perf/SLOs.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+The goal for a **Burst Test** is to evaluate the system's performance under heavy load when the user is most
+interested in performing an operation as quickly as possible.  A good example of a Burst workload is a sudden
+spike in demand for compute resources to handle a massive increase of users.
+
+The Burst Test measures batch VM/VMI creation latency so here's how KubeVirt tests it. The Burst Test will
+ramp up object count in a datacenter, measure, then ramp down object count in a datacenter. The measurement
+will include the Object Creation rate (objects/second), VMI Phase Transition Rate, and count Kubernetes REST API calls.
+
+The **Steady State Test** takes a different approach than the burst test in that instead of measuring how quickly
+a system can move between differant capacities, the Steady State Test focuses on how well the system can maintain
```
Comment (marceloamaral) [inline docs/scale-and-perf/SLOs.md:17]: differant -> different

File: docs/scale-and-perf/SLOs.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+max capacity or near max capacity.  For example, a system may create a certain number of warm resources
```
Comment (marceloamaral) [inline docs/scale-and-perf/SLOs.md:18]: I would say the goal is not to maintain maximum capacity, but to maintain a constant number of operations per second.

For example, 20 creates and deletes per second. The total number of objects in the system will depend on the object's lifetime...

Comment (rthallisey) [reply docs/scale-and-perf/SLOs.md:18]: I'm implying that.  No matter what the delete rate is, we're measuring the creation rate.  If 20 creates per second is the max create output at the test's scale, then that's what we'll expect.

**Comment by marceloamaral** [comment]
This is a good point, maybe this document should be in kubevirt/community

**Comment by rthallisey** [comment]
/close
Will revisit

**Pull request closed without merging**